File: Kotlin/Programmers/Level 3/가장 긴 팰린드롬.kt
import kotlin.math.max

class Solution {
    fun solution(s: String): Int {
        var answer = 1
        for (i in s.indices) {
            answer = max(answer, palindromeLength(s, i - 1, i + 1))
            answer = max(answer, palindromeLength(s, i, i + 1))
        }

        return answer
    }

    fun palindromeLength(s: String, left: Int, right: Int): Int {
        var l = left
        var r = right
        while (0 <= l && r < s.length && s[l] == s[r]) {
            l -= 1
            r += 1
        }

        return r - l - 1
    }
}

File: JavaScript/Programmers/Level 3/가장 긴 팰린드롬.js
function solution(s) {
    let answer = 1
    for (let i = 0; i < s.length; i++) {
        answer = Math.max(
            answer,
            palindromeLength(s, i, i + 1),
            palindromeLength(s, i - 1, i + 1)
        )
    }
    
    return answer
}

function palindromeLength(s, left, right) {
    let [l, r] = [left, right]
    while (0 <= l && r < s.length && s[l] === s[r]) {
        l -= 1
        r += 1
    }
    
    return r - l - 1
}